justfile

```
@@ -137,8 +137,14 @@ grafana:
     docker compose up grafana
 
 
+# Run a metrics task (defaults to running all tasks)
 metrics *args: devenv
-    $BIN/python -m metrics {{ args }}
+    #!/usr/bin/env bash
+    set -euo pipefail
+
+    MODULE="metrics.tasks"
+    test -z "{{ args }}" || MODULE="metrics.tasks.{{ args }}"
```

I had to read this several times. Could it be an if/else instead?

```suggestion
    if -n "{{ args }}"; then
        MODULE="metrics.tasks.{{ args }}"
    else
        MODULE="metrics.tasks"
    fi
```

---

Yes, happy for you to change it. I used `test` because that seemed to be the convention in our justfiles, but if it's not readable then happy for it to be changed.

---

for interest, you also have the option of just's own [if/else](
https://just.systems/man/en/chapter_34.html), like this:

```
  MODULE="metrics.tasts.{{ if args == "" { "" } else { "." + args } }}"
```

---

@madwort oh nice! Thanks for that suggestion. I've updated the justfile with that now.